lib/avro-patches/schema_validator/schema_validator.rb

```diff
@@ -64,15 +64,55 @@ def to_s
 
     class << self
       def validate!(expected_schema, datum)
-        result = Result.new
-        validate_recursive(expected_schema, datum, ROOT_IDENTIFIER, result)
-        fail ValidationError, result if result.failure?
-        result
+        with_result do |result|
+          validate_recursive(expected_schema, datum, ROOT_IDENTIFIER, result)
+        end
+      end
+
+      def validate_simple!(expected_schema, datum)
+        with_result do |result|
+          validate_simple(expected_schema, datum, ROOT_IDENTIFIER, result)
+        end
       end
 
       private
 
+      def with_result
+        result = Avro::SchemaValidator::Result.new
+        yield result
+        fail Avro::SchemaValidator::ValidationError, result if result.failure?
+        result
+      end
+
+      def validate_type(expected_schema)
+        unless Avro::Schema::VALID_TYPES_SYM.include?(expected_schema.type_sym)
+          fail "Unexpected schema type #{expected_schema.type_sym} #{expected_schema.inspect}"
+        end
+      end
+
       def validate_recursive(expected_schema, datum, path, result)
```

Comment:
Yes, they are. I found it easiest to think applying it to the existing layers of patches in this gem. This is how I'll have to submit it back to the original Avro project as the `SchemaValidator` is on master, but the logical type support is still in PR.